test/jobs/payout/uphold_job_test.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class UpholdJobTest < ActiveSupport::TestCase
+  test "it doesn't include Japanese payouts" do
+    uphold_in_japan = publishers(:uphold_in_japan)
+    assert uphold_in_japan.selected_wallet_provider.japanese_account?
+    uphold_job = Payout::UpholdJob.new
+    uphold_job.perform
+    assert IncludePublisherInPayoutReportJob.jobs.select { |job| job["args"].first["publisher_id"] == uphold_in_japan.id }.count, 0
+  end
+
+  test "it doesn't include Japanese payouts when providing ids" do
+    uphold_in_japan = publishers(:uphold_in_japan)
+    uphold_job = Payout::UpholdJob.new
+    uphold_job.perform(json_args: { publisher_ids: [uphold_in_japan.id] }.to_json)
+    assert IncludePublisherInPayoutReportJob.jobs.select { |job| job["args"].first["publisher_id"] == uphold_in_japan.id }.count, 0
+  end
+
+  test "for order and key presence" do
+    Sidekiq::Worker.clear_all
+    uphold_job = Payout::UpholdJob.new
+    uphold_job.perform(json_args: { payout_report_id: payout_reports(:one).id }.to_json)
+    visited_publisher_id = nil
+    assert IncludePublisherInPayoutReportJob.jobs.count > 1
+    IncludePublisherInPayoutReportJob.jobs.each do |job|
+      job_publisher_id = job["args"].first["publisher_id"]
+      assert visited_publisher_id <= job_publisher_id if visited_publisher_id.present?
```

Comment:
What if the `visited_publisher_id` is never present?

Reply:
Test checks it at the end

Reply:
It is compared to something which might be nil.

Reply:
It's checked at the end of the line with `if visited_publisher_id.present?`